### PR TITLE
Miscelleanous fixes.

### PR DESCRIPTION
**Pull request by Natureshadow**
Build system fixes
Code quality fixes (fixes compile time warnings and undefined behaviour)

Accompanied by a patch to xorgxrdp (did I mention the two should probably be
subtree-merged ;)?).

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -24,11 +24,13 @@ SUBDIRS = \
   mc \
   $(NEUTRINORDPDIR) \
   libxrdp \
+  librfxcodec \
```
Comment (proski) [inline Makefile.am]: Users who checkout xrdp, don't get librfxcodec by default. They would get a broken experience. That's really too big of a topic to be part of "miscellaneous fixes"

File: Makefile.am [hunk 2]
```diff
@@ -24,11 +24,13 @@ SUBDIRS = \
...
   xrdp \
   sesman \
   keygen \
   docs \
   instfiles \
   genkeymap \
   xrdpapi \
+  xorgxrdp \
```
Comment (proski) [inline Makefile.am]: Likewise, we need to decide if we want xorgxrdp to be part of the tree. I believe most users would want to use the `/usr` prefix for xorgxrdp to make it work, yet they make want to use `/usr/local` or other prefix for xrdp. I would rather have xorgxrdp as a completely separate package.

File: common/Makefile.am [hunk 3]
```diff
@@ -47,7 +47,12 @@ libcommon_la_SOURCES = \
   xrdp_rail.h \
   $(PIXMAN_SOURCES)
 
+if NEED_LIBDL
+EXTRA_LIBDL = -ldl
+endif
+
 libcommon_la_LIBADD = \
+  $(EXTRA_LIBDL) \
```
Comment (proski) [inline common/Makefile.am]: EXTRA_LIBDL is a poor name. I would prefer OPENSSL_LIBS, which would get `-ldl` added to it conditionally. A bigger issue is that we should support pkg-config for OpenSSL and take the settings from there.

File: configure.ac [hunk 4]
```diff
@@ -246,6 +246,14 @@ AC_CHECK_HEADER([X11/extensions/Xrandr.h], [],
   [AC_MSG_ERROR([please install libxrandr-dev or libXrandr-devel])],
   [#include <X11/Xlib.h>])
 
+# checking if we need libdl
```
Comment (proski) [inline configure.ac]: That message should be printed for users to show them what configure is doing. Again, this may not be needed if we use pkg-config to discover OpenSSL.

File: libxrdp/xrdp_jpeg_compress.c [hunk 5]
```diff
@@ -265,8 +265,8 @@ my_term_destination(j_compress_ptr cinfo)
 
 /*****************************************************************************/
 static int APP_CC
-jp_do_compress(char *data, int width, int height, int bpp, int quality,
-               char *comp_data, int *comp_data_bytes)
+jp_do_compress(JOCTET *data, int width, int height, int bpp, int quality,
```
Comment (proski) [inline libxrdp/xrdp_jpeg_compress.c]: If cleaning up this code, you may also want to add `const` to the pointer (unless it leads to a cascade of changes, in which case it would need to be a separate patch).

File: sesman/chansrv/chansrv_fuse.c [hunk 6]
```diff
@@ -265,7 +272,9 @@ static void xfuse_create_file(fuse_req_t req, fuse_ino_t parent,
 #endif
 
 static void xfuse_dump_fs(void);
+#if 0
```
Comment (proski) [inline sesman/chansrv/chansrv_fuse.c]: Please don't do it. It the code is not needed, please remove it. Don't let it rot. If the commented out code is needed (e.g. it illustrates something), please also add an explanation to the code.

File: sesman/chansrv/chansrv_fuse.c [hunk 7]
```diff
@@ -122,6 +128,7 @@ void xfuse_devredir_cb_file_close(void *vp)                                  {}
 #define LOG_ERROR   0
 #define LOG_INFO    1
 #define LOG_DEBUG   2
+#undef LOG_LEVEL
```
Comment (proski) [inline sesman/chansrv/chansrv_fuse.c]: Why do you need #undef here? Do you get a warning?

File: sesman/chansrv/chansrv_fuse.c [hunk 8]
```diff
@@ -1084,20 +1101,22 @@ static void xfuse_dump_fs()
...
  *****************************************************************************/
 
+#if 0
 static void xfuse_dump_xrdp_inode(struct xrdp_inode *xino)
 {
     log_debug("--- dumping struct xinode ---");
     log_debug("name:          %s", xino->name);
-    log_debug("parent_inode:  %ld", xino->parent_inode);
-    log_debug("inode:         %ld", xino->inode);
+    log_debug("parent_inode:  %ld", (long)xino->parent_inode);
```
Comment (proski) [inline sesman/chansrv/chansrv_fuse.c]: Why? Did you get a warning? What's your compiler? Unneeded casts are evil, then can hide issues.

File: sesman/chansrv/chansrv_fuse.c [hunk 9]
```diff
@@ -2850,9 +2873,9 @@ static void xfuse_cb_setattr(fuse_req_t req, fuse_ino_t ino, struct stat *attr,
 
     if (to_set & FUSE_SET_ATTR_SIZE)
     {
-        log_debug("previous file size: %d", attr->st_size);
+        log_debug("previous file size: %lld", (long long)attr->st_size);
```
Comment (proski) [inline sesman/chansrv/chansrv_fuse.c]: Why cannot you use the `z` modifier here? Please provide some context for your environment.

File: sesman/session.c [hunk 10]
```diff
@@ -617,7 +617,7 @@ session_start_fork(int width, int height, int bpp, char *username,
                 else
                 {
                     log_message(LOG_LEVEL_ERROR, "another Xserver might "
-                                "already be active on display %d - see log", display);
+                                "already be active on display %d - see log; possibly comment out the two param lines -logfile /dev/null in /etc/xrdp/sesman.ini to debug", display);
```
Comment (proski) [inline sesman/session.c]: I would rather have the log enabled by default, so that we don't have to show such instructions.

File: xrdp/Makefile.am [hunk 11]
```diff
@@ -19,7 +19,7 @@ endif
 if XRDP_RFXCODEC
 AM_CPPFLAGS += -DXRDP_RFXCODEC
 AM_CPPFLAGS += -I$(top_srcdir)/librfxcodec/include
-XRDP_EXTRA_LIBS += $(top_srcdir)/librfxcodec/src/.libs/librfxencode.a
+XRDP_EXTRA_LIBS += $(top_srcdir)/librfxcodec/src/.libs/librfxencode.la
```
Comment (proski) [inline xrdp/Makefile.am]: You don't want `.libs` here if using `*.la`, it was a hack to force static linking. Again, this is not a cleanup material.

**Comment by metalefty** [comment]
Some cleanups in this PR are useful but needs to be more separated.

File: docs/man/xrdp-dis.1 [hunk 12]
```diff
@@ -1,4 +1,4 @@
-.TH "xrdp-dis" "8" "0.7.0" "xrdp team"
+.TH "xrdp-dis" "1" "0.7.0" "xrdp team"
```
Comment (metalefty) [inline docs/man/xrdp-dis.1]: Fixed as #488.

File: libxrdp/xrdp_jpeg_compress.c (hunk 5 above)
Comment (metalefty) [inline libxrdp/xrdp_jpeg_compress.c]: Fixed as #503.

**Comment by metalefty** [comment]
The purpose of downstream patches should be disclosed.
Please do not just throw them to upstream. They need to be separated by purpose.

File: sesman/chansrv/devredir.h [hunk 13]
```diff
@@ -101,6 +101,8 @@ int dev_redir_file_open(void *fusep, tui32 device_id, char *path,
 
 int devredir_file_close(void *fusep, tui32 device_id, tui32 file_id);
 
+int APP_CC devredir_rmdir_or_file(void *fusep, tui32 device_id, char *path, int mode);
```
Comment (proski) [inline sesman/chansrv/devredir.h]: Fixed in #502
